### ISSUE TITLE
Conditional requirement doesn't work as expected

### ISSUE THREAD

**Issue by sbl03** [jzaefferer/jquery-validation#769]
HTML
````html
<input type="text" name="start_time" id="start_time" />
````
JS
````javascript
$("#event_form").validate({
    rules: {
        start_time: {
			required: function(element) {
				alert($("#start_time").val() == 'test');
				return $("#start_time").val() == 'test';
				//return $("#other_field").val() == 'test'; **works**
			}
		}
    }
}
````

I get an alert box at the appropriate events with the correct value, but it returns false regardless of the value in the textbox. I suspect this is because I am relying on the textbox value of itself, but I'm not sure if this is intentional or a bug. Furthermore, even though the alert is triggered with every keyUp, it only updates the validity when I call `valid()`.

**Comment by jzaefferer** [comment]
I'm sorry for the lack of activity on this issue. Instead of leaving it open any longer, I decided to close old issues without trying to address them, to longer give the false impression that it will get addressed eventually.

To the reporter (or anyone else interested in this issue): If you're affected by the same issue, consider opening a new issue, with a testpage that shows the issue. Even better, try to fix the issue yourself, and improve the project by sending a pull request. This may seem daunting at first, but you'll likely learn some useful skills that you can apply elsewhere as well. And you can help keep this project alive. [We've documented how to do these things, too](https://github.com/jzaefferer/jquery-validation/blob/master/CONTRIBUTING.md). A patch is worth a thousand issues!
